mitodl/mit-open
# ContentFiles for edx/ocw learning resources

#### Pre-Flight checklist



#### What are the relevant tickets?
Closes #18 

#### What's this PR do?
- Adds `ContentFile` model, serializers, views, urls including 3 endpoints:
  ```
  /api/v1/contentfiles/
  /api/v1/learning_resources/<resource_id>/contentfiles/
  /api/v1/courses/<resource_id>/contentfiles/
  ```  
- Moves the `backpopulate_xpro_files` management command from `course_catalog` to `learning_resources`
- Related: https://github.com/mitodl/salt-ops/pull/1583

#### How should this be manually tested?
Make sure the following are populated in your .env:
```
XPRO_LEARNING_COURSE_BUCKET_NAME=<same as heroku open-discussions-rc value>
XPRO_CATALOG_API_URL=https://xpro.mit.edu/api/programs/
XPRO_COURSES_API_URL=https://xpro.mit.edu/api/courses/
AWS_ACCESS_KEY_ID=<same as heroku mit-open-rc value>
AWS_SECRET_ACCESS_KEY=<same as heroku mit-open-rc value>
```

Run the following:
```
manage.py backpopulate_xpro_data
manage.py backpopulate_xpro_files
```
Both should complete successfully.

Try the following urls, they should all work and return results:
```
http://localhost:8063/api/v1/contentfiles/
http://localhost:8063/api/v1/contentfiles/?run=<some_run__id>/
http://localhost:8063/api/v1/learning_resources/<xpro_resource_id>/
http://localhost:8063/api/v1/learning_resources/<xpro_resource_id>/?run=<some_run__id>/
http://localhost:8063/api/v1/courses/<xpro_resource_id>/
http://localhost:8063/api/v1/courses/<xpro_resource_id>/?run=<some_run__id>/
```
